src/Elastic.Apm.Elasticsearch/ElasticsearchDiagnosticsSubscriber.cs

```diff
@@ -15,6 +15,10 @@ public class ElasticsearchDiagnosticsSubscriber : IDiagnosticsSubscriber
 		public IDisposable Subscribe(IApmAgent agentComponents)
 		{
 			var composite = new CompositeDisposable();
+
+			if (!agentComponents.ConfigurationReader.Enabled)
```

Comment:
Wonder if a base class implementation should be introduced, that does this check, and then calls into an overridden method that sets up the specific subscribers for the derived diagnostics subscriber. With such a base class, the behaviour can be enforced in one place.

Reply:
Sounds like a good idea. Opened an issue for this: https://github.com/elastic/apm-agent-dotnet/issues/1018/.